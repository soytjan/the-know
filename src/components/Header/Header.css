@import url("https://fonts.googleapis.com/css?family=Open+Sans:300,400|Poiret+One");
.Header {
  display: flex;
  flex-direction: row;
  font-family: "Poiret One", cursive;
  justify-content: space-between;
  border-bottom: 10px solid #4ECDC4; }
  .Header h1 {
    font-size: 2.5em;
    padding: 15px 15px 10px; }
  .Header a {
    text-decoration: none;
    color: #383838;
    font-size: 1.5em; }
    .Header a:hover {
      border-bottom: 3px solid #BEECE9; }

.header-link-cont {
  display: flex; }

.header-link-box {
  border-left: 1px solid #383838;
  display: flex;
  justify-content: center;
  align-items: center;
  width: 150px;
  height: 100%; }

@media screen and (max-width: 600px) {
  .Header {
    flex-direction: column;
    align-items: center;
    padding-bottom: 15px;
    background-color: #FBFCFC; }
  .header-link-box {
    border: 1px solid #383838;
    padding: 10px; } }
