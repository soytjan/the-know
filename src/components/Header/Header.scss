@import "../../variables.scss";

.Header {
  display: flex;
  flex-direction: row;
  font-family: $header-font;
  justify-content: space-between;
  border-bottom: 10px solid $turq;

  h1 {
    font-size: 2.5em;
  }

  a {
    text-decoration: none;
    color: $default-font-color;
    font-size: 1.5em;

    &:hover {
      background-color: $headerlink-hover-color;
    }
  }
}

.header-link-box {
  border-left: 1px solid $black;
  display: inline-block;
  width: 120px;
  height: 100%;
}