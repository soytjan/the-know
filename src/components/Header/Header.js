import React from 'react';
import { NavLink } from 'react-router-dom';
import './Header.css';

// add in NavLink tags but will need to set up the routes
const Header = () => {
  return (
    <header className="Header">
      <h1>THE KNOW</h1>
      <div className='header-link-cont'>
        <div className='header-link-box'>
          <NavLink to='/signup'>SIGN UP</NavLink>
        </div>
        <div className='header-link-box'>
          <NavLink to='/login'>LOGIN</NavLink>
        </div>
      </div>
    </header>
  )
}

export default Header;