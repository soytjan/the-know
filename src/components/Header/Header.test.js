import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header';

describe('Header', () => {
  it('should match snapshot', () => {
    const renderedComponent = shallow(<Header />);

    expect(renderedComponent).toMatchSnapshot();
  })
})