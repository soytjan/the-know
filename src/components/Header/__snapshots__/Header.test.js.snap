// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    to="/home"
          >
                    THE KNOW
          </NavLink>
</h1>,
        <div
          className="header-link-cont"
>
          <div
                    className="header-link-box"
          >
                    <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              to="/home/favorites"
                    >
                              FAVORITES
                    </NavLink>
          </div>
</div>,
      ],
      "className": "Header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <NavLink
            activeClassName="active"
            ariaCurrent="true"
            to="/home"
>
            THE KNOW
</NavLink>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "THE KNOW",
            "to": "/home",
          },
          "ref": null,
          "rendered": "THE KNOW",
          "type": [Function],
        },
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="header-link-box"
>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        to="/home/favorites"
            >
                        FAVORITES
            </NavLink>
</div>,
          "className": "header-link-cont",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="active"
              ariaCurrent="true"
              to="/home/favorites"
>
              FAVORITES
</NavLink>,
            "className": "header-link-box",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "FAVORITES",
              "to": "/home/favorites",
            },
            "ref": null,
            "rendered": "FAVORITES",
            "type": [Function],
          },
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        to="/home"
            >
                        THE KNOW
            </NavLink>
</h1>,
          <div
            className="header-link-cont"
>
            <div
                        className="header-link-box"
            >
                        <NavLink
                                    activeClassName="active"
                                    ariaCurrent="true"
                                    to="/home/favorites"
                        >
                                    FAVORITES
                        </NavLink>
            </div>
</div>,
        ],
        "className": "Header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="active"
              ariaCurrent="true"
              to="/home"
>
              THE KNOW
</NavLink>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "THE KNOW",
              "to": "/home",
            },
            "ref": null,
            "rendered": "THE KNOW",
            "type": [Function],
          },
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="header-link-box"
>
              <NavLink
                            activeClassName="active"
                            ariaCurrent="true"
                            to="/home/favorites"
              >
                            FAVORITES
              </NavLink>
</div>,
            "className": "header-link-cont",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="active"
                ariaCurrent="true"
                to="/home/favorites"
>
                FAVORITES
</NavLink>,
              "className": "header-link-box",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "FAVORITES",
                "to": "/home/favorites",
              },
              "ref": null,
              "rendered": "FAVORITES",
              "type": [Function],
            },
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
